entities.py

```diff
@@ -0,0 +1,194 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+
+class Position(object):
+
+    def __init__(self, x, y):
+        self.x = x
+        self.y = y
+
+    def move(self, x, y):
+        self.x += x
+        self.y += y
+
+    def __repr__(self):
+        return "(%d, %d)" % (self.x, self.y)
+
+
+STARTING_PLAYER_POS = Position(0, 0)
+
+
+class ProgrammingError(Exception):
+    pass
+
+
+class Entity(object):
+
+    _pool = []
+    _id = 0
+
+    def __init__(self, name, display_name, pos, type):
+        # FIXME: find a better heuristic
+        for entity in Entity._pool:
+            if entity.name == name:
+                raise ProgrammingError
+        # assign it a unique id
+        self.id = Entity._id
+        Entity._id += 1
+        # other stuff
+        self.name = name + '_%d' % self.id
```

Comment:
Isn't `name` redundant since we have `id` and `type` variables which already identify the entity?